src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java

```diff
@@ -206,4 +206,19 @@ private void sendAvailableNotice(CheckInProcessRecords records, Request firstReq
     }
     return succeeded(records);
   }
+
+  public CheckInProcessRecords setInHouseUse(CheckInProcessRecords checkInProcessRecords) {
+    return checkInProcessRecords
+      .withInHouseUse(isInHouseUse(checkInProcessRecords));
+  }
+
+  private boolean isInHouseUse(CheckInProcessRecords records) {
+    if (records.getItem().getLocation() == null) {
+      return false;
+    }
+
+    return records.getItem().isAvailable()
+      && (records.getRequestQueue() == null || records.getRequestQueue().size() == 0)
+      && records.getItem().getLocation().homeLocationIsServedBy(records.getCheckInServicePointId());
```

Comment:
I'd like to start improving the balance between API tests and unit tests, could this be extracted and made so that unit tests can express all of the conditions for in house use?

Reply:
This method is being called in `setInHouseUse` which is a public class. 

Do we need unit tests for this class?

Reply:
> This method is being called in `setInHouseUse` which is a public class.

I don't follow your reasoning, why does that affect this decision? Or are you suggesting that the public method should be included in the extraction as well?

I don't think so, because the intent of the process adapter is to clean up the edges between steps in the process.

> Do we need unit tests for this class?

I think that mod-circulation is too reliant upon the API tests to check most of the behaviour in the system. These tests are expensive to maintain and execute.

I think we need to start moving more of the behaviour testing to smaller scoped tests. Meaning we will need fewer API tests to demonstrate the high level flows through the system.

Does that make sense? What do you think about this intent?

Reply:
I mean that if we want to unit test the `isInHouseUse` method we can do it by executing `setInHouseUse` method which sets the result of `isInHouseUse` to the CheckInProcessRecords object.

Reply:
> I mean that if we want to unit test the `isInHouseUse` method we can do it by executing `setInHouseUse` method which sets the result of `isInHouseUse` to the CheckInProcessRecords object.

Ok. I think the logic could be more isolated than that, however if you believe that is the best seam to design the unit tests around then I am ok with that.

I think the logic needs to be separated from the process adapter, in either case, to keep the responsibilities separate.

Reply:
The smaller scale tests might also help to cover some of the parts of this logic that Sonar is reporting as not covered.

Reply:
Do you have any idea where we can put this logic? Should I create a new service class for this or reuse existing one?

Reply:
Maybe `org.folio.circulation.domain.LoanCheckInService`?

Reply:
Sure, or something specific to in-house use